devicetypes/smartthings/fibaro-double-switch.src/fibaro-double-switch.groovy

```diff
@@ -0,0 +1,786 @@
+			eventMap.name = "energy"
+			eventMap.value = cmd.scaledMeterValue
+			eventMap.unit = "kWh"
+		} else if (cmd.scale == 2) {
+			eventMap.name = "power"
+			eventMap.value = Math.round(cmd.scaledMeterValue)
+			eventMap.unit = "W"
+		}
+	}
+	eventMap
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, ep) {
+	log.warn "Unhandled ${cmd}" + (ep ? " from endpoint $ep" : "")
+}
+
+def on() {
+	onOffCmd(0xFF)
+}
+
+def off() {
+	onOffCmd(0x00)
+}
+
+def ping() {
+	refresh()
+}
+
+def childOnOff(deviceNetworkId, value) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
+}
+
+private onOffCmd(value, endpoint = 1) {
+	delayBetween([
+			encap(zwave.basicV1.basicSet(value: value), endpoint),
+			encap(zwave.basicV1.basicGet(), endpoint),
+			"delay 3000",
+			encap(zwave.meterV3.meterGet(scale: 0), endpoint),
+			encap(zwave.meterV3.meterGet(scale: 2), endpoint)
+	])
+}
+
+private refreshAll(includeMeterGet = true) {
+
+	def endpoints = [1]
+
+	childDevices.each {
+		def switchId = getSwitchId(it.deviceNetworkId)
+		if (switchId != null) {
+			endpoints << switchId
+		}
+	}
+	sendHubCommand refresh(endpoints,includeMeterGet)
+}
+
+def childRefresh(deviceNetworkId, includeMeterGet = true) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		sendHubCommand refresh([switchId],includeMeterGet)
+	}
+}
+
+def refresh(endpoints = [1], includeMeterGet = true) {
+	def cmds = []
+	endpoints.each {
+		cmds << [encap(zwave.basicV1.basicGet(), it)]
+		if (includeMeterGet) {
+			cmds << encap(zwave.meterV3.meterGet(scale: 0), it)
+			cmds << encap(zwave.meterV3.meterGet(scale: 2), it)
+		}
+	}
+	delayBetween(cmds, 200)
+}
+
+private resetAll() {
+	childDevices.each { childReset(it.deviceNetworkId) }
+	sendHubCommand reset()
+}
+
+def childReset(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		log.debug "Child reset switchId: ${switchId}"
+		sendHubCommand reset(switchId)
+	}
+}
+
+def reset(endpoint = 1) {
+	log.debug "Resetting endpoint: ${endpoint}"
+	delayBetween([
+			encap(zwave.meterV3.meterReset(), endpoint),
+			encap(zwave.meterV3.meterGet(scale: 0), endpoint),
+			"delay 500"
+	], 500)
+}
+
+def getSwitchId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+		log.debug "encap command: ${cmd} "
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private addChildSwitches(numberOfSwitches) {
+	for (def endpoint : 2..numberOfSwitches) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName + "-${endpoint}"
```

Comment:
I think the component name would look a little cleaner as `device.displayName + " ${endpoint}"`.

Reply:
There's supposed to be a space before the `$endpoint` part. It appears that the formatting didn't treat that space as non-breaking.

```
device.displayName + " ${endpoint}"
```

Reply:
ok, fixed